File: eFormAPI/Plugins/ItemsGroupPlanning.Pn/ItemsGroupPlanning.Pn/Infrastructure/Models/ItemsListPnModel.cs
namespace ItemsGroupPlanning.Pn.Infrastructure.Models
{
    using System;
    using System.Collections.Generic;
    using Microting.ItemsGroupPlanningBase.Infrastructure.Enums;

    public class ItemsListPnModel
    {
        public int Id { get; set; }
        
        public string Name { get; set; }
        
        public string Description { get; set; }
        
        public int RepeatEvery { get; set; }
        
        public RepeatType RepeatType { get; set; }
        
        public DateTime? RepeatUntil { get; set; }
        
        public DayOfWeek? DayOfWeek { get; set; }
        
        public int? DayOfMonth { get; set; }
        
        public DateTime? LastExecutedTime { get; set; }
        
        public int RelatedEFormId { get; set; }
        
        public string RelatedEFormName { get; set; }
        
        public bool LabelEnabled { get; set; }
        
        public bool DescriptionEnabled { get; set; }
        
        public bool DeployedAtEnabled { get; set; }

        public bool DoneAtEnabled { get; set; }

        public bool DoneByUserNameEnabled { get; set; }

        public bool UploadedDataEnabled { get; set; }
        
        public bool ItemNumberEnabled { get; set; }
        
        public bool LocationCodeEnabled { get; set; }
        
        public bool BuildYearEnabled { get; set; }
        
        public bool TypeEnabled { get; set; }
        
        public bool NumberOfImagesEnabled { get; set; }
        
        public string FieldName1 { get; set; }

        public string FieldName2 { get; set; }

        public string FieldName3 { get; set; }

        public string FieldName4 { get; set; }

        public string FieldName5 { get; set; }

        public string FieldName6 { get; set; }

        public string FieldName7 { get; set; }

        public string FieldName8 { get; set; }

        public string FieldName9 { get; set; }

        public string FieldName10 { get; set; }
        
        public int? SdkFieldId1 { get; set; }
        
        public int? SdkFieldId2 { get; set; }
        
        public int? SdkFieldId3 { get; set; }
        
        public int? SdkFieldId4 { get; set; }
        
        public int? SdkFieldId5 { get; set; }
        
        public int? SdkFieldId6 { get; set; }
        
        public int? SdkFieldId7 { get; set; }
        
        public int? SdkFieldId8 { get; set; }
        
        public int? SdkFieldId9 { get; set; }
        
        public int? SdkFieldId10 { get; set; }

        public List<ItemsListPnItemModel> Items { get; set; }
            = new List<ItemsListPnItemModel>();
    }
}

File: eform-client/src/app/plugins/modules/items-group-planning-pn/models/list/items-list-pn.model.ts
import {ItemsListPnItemModel} from './items-list-pn-item.model';
import {Moment} from 'moment';

export class ItemsListsPnModel {
  total: number;
  lists: Array<ItemsListPnModel> = [];
}

export class ItemsListPnModel {
  id: number;
  name: string;
  description: string;
  repeatEvery: number;
  repeatType: number;
  dayOfWeek: number;
  dayOfMonth: number;
  repeatUntil: string;
  internalRepeatUntil: string;
  relatedEFormId: number;
  relatedEFormName: string;
  items: ItemsListPnItemModel[] = [];
  deployedAtEnabled: boolean;
  doneAtEnabled: boolean;
  doneByUserNameEnabled: boolean;
  uploadedDataEnabled: boolean;
  labelEnabled: boolean;
  descriptionEnabled: boolean;
  sdkFieldEnabled1: boolean;
  sdkFieldEnabled2: boolean;
  sdkFieldEnabled3: boolean;
  sdkFieldEnabled4: boolean;
  sdkFieldEnabled5: boolean;
  sdkFieldEnabled6: boolean;
  sdkFieldEnabled7: boolean;
  sdkFieldEnabled8: boolean;
  sdkFieldEnabled9: boolean;
  sdkFieldEnabled10: boolean;
  itemNumberEnabled: boolean;
  locationCodeEnabled: boolean;
  buildYearEnabled: boolean;
  typeEnabled: boolean;
  numberOfImagesEnabled: boolean;
  sdkFieldId1: number;
  sdkFieldId2: number;
  sdkFieldId3: number;
  sdkFieldId4: number;
  sdkFieldId5: number;
  sdkFieldId6: number;
  sdkFieldId7: number;
  sdkFieldId8: number;
  sdkFieldId9: number;
  sdkFieldId10: number;
}
